### PR TITLE
Using windows_reboot for all powershell versions

### PR DESCRIPTION
**Pull request by NimishaS**
Powershell recipes for all versions should reboot machine after installing powershell.  Issue: https://github.com/chef-windows/powershell/issues/14#issuecomment-65636097  

Note: Reboot happens node['powershell']['installation_reboot_mode'] = 'immediate_reboot' in default attributes.

### CODE REVIEW

File: recipes/powershell5.rb [hunk 1]
```diff
@@ -31,11 +33,18 @@
...
+    powershell_version = shell_out.stdout.to_f
+
+    # Reboot if user specifies immediate_reboot
+    include_recipe 'powershell::windows_reboot' if node['powershell']['installation_reboot_mode'] == 'immediate_reboot' && powershell_version < 5
+
     windows_package 'Windows Management Framework Core 5.0' do
       source node['powershell']['powershell5']['url']
       checksum node['powershell']['powershell5']['checksum']
       installer_type :custom
-      options '/quiet /norestart'
```
Comment (adamedx) [inline recipes/powershell5.rb]: Why are we removing these flags?

Comment (NimishaS) [inline recipes/powershell5.rb]: I should have removed only `/norestart` as we need to restart. I'll make required changes.

File: recipes/powershell5.rb [hunk 2]
```diff
@@ -31,11 +33,19 @@
...
+      options '/quiet'
       action :install
       not_if { registry_data_exists?('HKLM\SOFTWARE\Microsoft\PowerShell\3\PowerShellEngine', { :name => 'PowerShellVersion', :type => :string, :data => '5.0.9701.0' }) }
```
Comment (adamedx) [inline recipes/powershell5.rb]: Is this the right regkey? For powershell 5, wouldn't we be checking `\\powershell\\5` instead of `\\powershell\\3`?

File: recipes/powershell4.rb [hunk 3]
```diff
@@ -21,6 +21,8 @@
 # PowerShell 4.0 Download Page
 # http://www.microsoft.com/en-us/download/details.aspx?id=40855
```
Comment (adamedx) [inline recipes/powershell4.rb]: See the comment below about `\\powershell\\3` in the registry (not part of this commit) -- shouldn't we be checking for `4`?

File: recipes/powershell2.rb [hunk 4]
```diff
@@ -61,11 +63,19 @@
 
     include_recipe 'ms_dotnet2'
 
+    powershell_version_cmd = 'powershell.exe $PSVersionTable.PSVersion.Major'
+    shell_out = Mixlib::ShellOut.new(powershell_version_cmd)
+    shell_out.run_command
+    powershell_version = shell_out.stdout.to_f
+
+    # Reboot if user specifies immediate_reboot
```
Comment (adamedx) [inline recipes/powershell2.rb]: So to be clear, we always reboot, no matter what? If that's the case, we need to update the README.md to make that clear.

Separate thing in the `windows_reboot` recipe -- we have a guard expression that is always `false` -- if that's the case, we should remove the guard altogether.

File: recipes/powershell5.rb (hunk 2 above)
Comment (NimishaS) [inline recipes/powershell5.rb]: @adamedx , this is correct. There are only two folders that exist in registery: `\\powershell\\1` and `\\powershell\\3`. Registery data for powershell2 is stored in `\\powershell\\1` and for powershell3, powershell4 and powershell5, it's stored in `\\powershell\\3`. 
They don't depict the powershell version. Version is depicted by the `data` field. e.g. in case of powershell5, its `:data => '5.0.9701.0'`

File: recipes/powershell2.rb (hunk 4 above)
Comment (NimishaS) [inline recipes/powershell2.rb]: @adamedx , we don't reboot in all cases. We reboot only if user specifies `default['powershell']['installation_reboot_mode'] == 'immediate_reboot'` in default attribute file.

Comment (adamedx) [inline recipes/powershell2.rb]: Right, but if they set this attribute, the ideal behavior is that it only reboots if a reboot is needed. Can't we make this conditional on the `reboot_required?` method?

**Comment by NimishaS** [comment]
Covered these fixes as a part of https://github.com/opscode-cookbooks/powershell/pull/38
